### PR TITLE
Fix Send to Active Notebook functionality

### PR DESCRIPTION
**Pull request by mofojed**
- Added `getLastUsedPanelOfTypes` to PanelManager (will be needed for Enterprise)
- Clean up some of the TypeScript types

### CODE REVIEW

File: packages/dashboard-core-plugins/src/ConsolePlugin.tsx [hunk 1]
```diff
@@ -442,26 +437,14 @@ export const ConsolePlugin = ({
...
-        (FileExplorerPanel as unknown) as ComponentType
-      ),
-      registerComponent(
-        LogPanel.COMPONENT,
-        (LogPanel as unknown) as ComponentType
-      ),
-      registerComponent(
-        NotebookPanel.COMPONENT,
-        (NotebookPanel as unknown) as ComponentType
+        (FileExplorerPanel as unknown) as PanelComponentType
```
Comment (mofojed) [inline packages/dashboard-core-plugins/src/ConsolePlugin.tsx:444]: @mattrunyon any idea how to get TypeScript typings correct so we don't need to explicitly convert this one? I'm having a hard time getting TypeScript to co-operate here.

Comment (mofojed) [reply packages/dashboard-core-plugins/src/ConsolePlugin.tsx:444]: The error:
```
Argument of type 'ConnectedComponent<typeof FileExplorerPanel, Omit<ClassAttributes<FileExplorerPanel> & PanelProps & { localDashboardId: string; metadata?: Record<...> | undefined; } & StateProps & DispatchProp<...>, "dispatch" | ... 2 more ... | "fileStorage"> & PanelProps & { ...; }>' is not assignable to parameter of type 'PanelComponentType<ComponentClass<{}, any>, Omit<ClassAttributes<FileExplorerPanel> & PanelProps & { localDashboardId: string; metadata?: Record<...> | undefined; } & StateProps & DispatchProp<...>, "dispatch" | ... 2 more ... | "fileStorage"> & PanelProps & { ...; }>'.
  Type 'ConnectedComponent<typeof FileExplorerPanel, Omit<ClassAttributes<FileExplorerPanel> & PanelProps & { localDashboardId: string; metadata?: Record<...> | undefined; } & StateProps & DispatchProp<...>, "dispatch" | ... 2 more ... | "fileStorage"> & PanelProps & { ...; }>' is not assignable to type 'FunctionComponent<{}>'.
    Types of parameters 'props' and 'props' are incompatible.
      Type '{ children?: ReactNode; }' is not assignable to type 'Omit<ClassAttributes<FileExplorerPanel> & PanelProps & { localDashboardId: string; metadata?: Record<string, unknown> | undefined; } & StateProps & DispatchProp<...>, "dispatch" | ... 2 more ... | "fileStorage"> & PanelProps & { ...; }'.
        Type '{ children?: ReactNode; }' is missing the following properties from type 'Omit<ClassAttributes<FileExplorerPanel> & PanelProps & { localDashboardId: string; metadata?: Record<string, unknown> | undefined; } & StateProps & DispatchProp<...>, "dispatch" | ... 2 more ... | "fileStorage">': glContainer, glEventHub, localDashboardIdts(2345)
```

File: packages/dashboard/src/PanelManager.test.ts [hunk 2]
```diff
@@ -72,3 +72,24 @@ it('gets last used panel of type properly', () => {
...
+  const panelB = new TestComponentB();
+  layout.eventHub.emit(PanelEvent.MOUNT, panelA);
+  layout.eventHub.emit(PanelEvent.MOUNT, panelB);
+
+  expect(panelManager.getLastUsedPanelOfTypes([TestComponentA])).toBe(panelA);
+  expect(
+    panelManager.getLastUsedPanelOfTypes([TestComponentA, TestComponentB])
+  ).toBe(panelB);
+  expect(
+    panelManager.getLastUsedPanelOfTypes([TestComponentB, TestComponentB])
```
Comment (vbabich) [inline packages/dashboard/src/PanelManager.test.ts:90]: I made a typo in the original code, this was supposed to be
```suggestion
    panelManager.getLastUsedPanelOfTypes([TestComponentB, TestComponentA])
```

File: packages/dashboard/src/PanelManager.ts [hunk 3]
```diff
@@ -162,8 +162,22 @@ class PanelManager {
...
+  }
+
+  getLastUsedPanelOfTypes(
+    types: PanelComponentType[]
+  ): PanelComponent | undefined {
+    return this.getLastUsedPanel(panel =>
+      types.some(
+        type =>
+          panel instanceof type ||
+          (isWrappedComponent(type) &&
```
Comment (vbabich) [inline packages/dashboard/src/PanelManager.ts:176]: Can we get away with a simple `panel => panel instanceof type` check here? just pass unwrapped components to `getLastUsedPanelOfTypes` and skip the `isWrappedComponent` check? I didn't think of this solution when I added this on the enterprise side, then I saw you had it simplified in DHC. Then I guess remove `| WrappedComponentType<C, P>` from the `PanelComponentType` definition too.

**Review by vbabich — CHANGES_REQUESTED** [review]

**Review by vbabich — APPROVED** [review]
Nice!

File: packages/dashboard/src/PanelManager.ts (hunk 3 above)
Comment (mofojed) [reply packages/dashboard/src/PanelManager.ts:181]: I've fixed this, but for posterity, we still need the WrappedComponentType for registerComponent anyway, so may as well make it work right!

**Pull request merged**